### PR TITLE
feat(DevTools): Allow to display the type of property in ControlDettails

### LINKED ISSUE #5883: [DevTools] Add Property Type Column on Properties DataGrid
In version 0.10.3 the Type column displays the current value type assigned to the property.

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Allow to display  the type of property in DevTools ControlDettails.

## What is the current behavior?

The Type column of ControlDettails  display  the type of current value of property.

immagine

## What is the updated/expected behavior with this PR?

- Renamed Type property in AssignedType
- Add PropertyType property
- Allow decode typename of nullable and generinc type

### Single Column (Default)

immagine

### Split Columns

immagine

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Fixes #5883

### CODE REVIEW

**Comment by maxkatz6** [comment]
It should be optional. Maybe add some toggle to View menu to enable or disable this column?
I would say disabled by default.

**Comment by workgroupengineering** [comment]
> 
> 
> It should be optional. Maybe add some toggle to View menu to enable or disable this column?
> I would say disabled by default.

What do you think if I add an option in DevToolsOptions?

**Comment by maxkatz6** [comment]
@workgroupengineering probably no. It's temporal view option. That might be added to DevToolsOptions, but more important to see it in the View menu.

**Comment by workgroupengineering** [comment]
> 
> 
> @workgroupengineering probably no. It's temporal view option. That might be added to DevToolsOptions, but more important to see it in the View menu.
> 
> Which means, I don't think many actually need it by default. Still, it definitely needed in some specific scenarios.

I tried but is blocked from issue #6282

**Comment by maxkatz6** [comment]
@workgroupengineering it's a bug, but it's possible to toggle visibility from code behind IIRC.

**Comment by workgroupengineering** [comment]
> 
> 
> @workgroupengineering it's a bug, but it's possible to toggle visibility from code behind IIRC.

Can you explain me how to do it?

**Comment by maxkatz6** [comment]
Read columns from data grid, find correct one by header, and change IsVisibility property.
But, yeah, we are in Avalonia repo, we are able to fix https://github.com/AvaloniaUI/Avalonia/issues/6282 instead.

**Comment by pr8x** [comment]
Couldn't we add the derived type in brackets after the base type and continue to call this column "Type"?

**Comment by MarchingCube** [comment]
> Couldn't we add the derived type in brackets after the base type and continue to call this column "Type"?

I wonder if this won't make sorting a bit weird.

**Comment by grokys** [comment]
> > Couldn't we add the derived type in brackets after the base type and continue to call this column "Type"?
> I wonder if this won't make sorting a bit weird.

This is what VS does in its watch views, so there's precedent. Also, who's sorting on Type anyway?

image

I think I'd prefer this as the existing columns often don't fit in the space anyway.

**Comment by workgroupengineering** [comment]
> I think I'd prefer this as the existing columns often don't fit in the space anyway.

The problem with the solution you suggest is the difficult sorting.

**Comment by pr8x** [comment]
@workgroupengineering As he mentioned sorting alphabetically by type really rarely happens (if at all) and should not be a priority for designing this.

**Comment by workgroupengineering** [comment]
> 
> 
> @workgroupengineering As he mentioned sorting alphabetically by type really rarely happens (if at all) and should not be a priority for designing this.

I am not agrre completely.
The purposes of this PR are:
- quickly identify the type of property
- quickly identify the type of the value assigned to the property

Merge into one column I don't think it meets the requirements.

Sorting is an extra feature that helps improve readability.

File: src/Avalonia.Diagnostics/Diagnostics/Views/MainView.xaml [hunk 1]
```diff
@@ -15,6 +15,16 @@
                       IsEnabled="False" />
           </MenuItem.Icon>
         </MenuItem>
+        <MenuItem Header="_Dettails">
```
Comment (maxkatz6) [inline src/Avalonia.Diagnostics/Diagnostics/Views/MainView.xaml:18]:
```suggestion
        <MenuItem Header="_Details">
```

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**